prov/shm/src/smr_progress.c

```diff
@@ -569,29 +569,47 @@ static int smr_progress_ipc(struct smr_cmd *cmd, enum fi_hmem_iface iface,
 }
 
 static void smr_do_atomic(void *src, void *dst, void *cmp, enum fi_datatype datatype,
-			  enum fi_op op, size_t cnt, uint16_t flags)
+			  enum fi_op op, size_t cnt, uint16_t flags, enum fi_hmem_iface iface, uint64_t device)
 {
 	char tmp_result[SMR_INJECT_SIZE];
 
 	if (ofi_atomic_isswap_op(op)) {
-		ofi_atomic_swap_handler(op, datatype, dst, src, cmp,
-					tmp_result, cnt);
+		if (iface == FI_HMEM_SYSTEM) {
+			ofi_atomic_swap_handler(op, datatype, dst, src, cmp, tmp_result, cnt);
+		} else {
+			char tmp_dst[SMR_INJECT_SIZE];
```

Comment:
Declare variables at the top of the function.  That, too, helps with readability.

Reply:
done